ticketbuyer/buyer.go

```diff
@@ -820,8 +838,14 @@ func (t *TicketPurchaser) Purchase(height int64) (*PurchaseStats, error) {
 		return ps, err
 	}
 
-	// Ticket purchase requires 2 blocks to confirm
-	expiry := int32(int(height) + t.ExpiryDelta() + 2)
+	expiryDelta := t.ExpiryDelta()
+	expiry := int32(0)
+
+	if expiryDelta != 0 {
+		// Ticket purchase requires 2 blocks to confirm
+		expiry = int32(int(height) + t.ExpiryDelta() + 2)
```

Comment:
`t.ExpiryDelta()` which is `expiryDelta` would be 0 at this point, any reason why you're not using `expiry = int32(int(height) +  2)` ?